test/powershell/Language/Scripting/CmdletDeclaration.Tests.ps1

```diff
@@ -0,0 +1,122 @@
+﻿Describe "Cmdlet declaration statement" -Tags "CI" {
+
+    $testData = @(
+        @{ Name = 'Verify non-cmdlet formatted names are allowed';
+           Script = '
+                function foo
+                {
+                    [CmdletBinding()]
+                    param()
+
+                    $a
+                }' },
+                @{ Name = 'Valid cmdlet names case 1';
+                Script = '
+                function foo
+                {
+                    [CmdletBinding()]
+                    param()
+
+                    $a
+                }' },
+                @{ Name = 'Valid cmdlet names case 2';
+                Script = '
+                function get-foo
+                {
+                    [CmdletBinding()]
+                    param()
+
+                    $a
+                }' },
+                @{ Name = 'Valid cmdlet names case 3';
+                Script = '
+                {
+                    [CmdletBinding()]
+                    param()
+
+                    $a
+                }' },
+                @{ 
+                Name ='Using parameter annotation in script cmdlets';                
+                Script = '
+                function get-foo
+                {
+                    param([Parameter(mandatory=$true)]$a)
+                    $a
+                }'},
+                @{
+                Name = 'Cmdlet declaration parameter: should process';
+                Script = '
+                function get-foo
+                {
+                    [CmdletBinding(SupportsShouldProcess=$true)]
+                    param([Parameter(mandatory=$true)]$a)
+                    $a
+                }' },
+                @{
+                Name = 'Cmdlet declaration parameter: confirm impact';
+                Script = '
+                function get-foo
+                {
+                    [CmdletBinding(SupportsShouldProcess=$true, ConfirmImpact="low")]
+                    param([Parameter(mandatory=$true)]$a)
+                    $a
+                }'},
+                @{
+                Name = 'Cmdlet declaration parameter: defaultparametersetname';
+                Script = '
+                function get-foo
+                {
+                    [CmdletBinding(SupportsShouldProcess=$true, ConfirmImpact="low", defaultparametersetname="set1")]
+                    param([Parameter(mandatory=$true)]$a)
+                    $a
+                }'},
+                @{
+                Name = 'cmdlet declaration within as cmdlet';
+                Script = '
+                function get-foo {
+                    [CmdletBinding()]
+                    param()
+
+	            function get-bar
+                    {
+                        param([Parameter(mandatory=$true)]$a)
+                        $a
+                    }
+                }'}
+                @{
+                Name = 'cmdlet declaration within scripblock';
```

Comment:
fixed

---
In reply to: [76495751](https://github.com/powershell/powershell/pull/2086#discussion_r76495751) [](ancestors = 76495751)